tests/Api/Serializer/RestJsonSerializerTest.php

```diff
@@ -91,6 +108,48 @@ public function testPreparesRequestsWithBlobButNoForcedContentType()
         $this->assertEquals('', $request->getHeaderLine('Content-Type'));
     }
 
+    public function testPreparesRequestsWithJsonValueTraitString()
+    {
+        $jsonValueArgs = '{"a":"b"}';
+        $request = $this->getRequest('foobar', ['baz' => $jsonValueArgs]);
+        $this->assertEquals('IntcImFcIjpcImJcIn0i', $request->getHeaderLine('baz'));
+        $this->assertEquals('POST', $request->getMethod());
+        $this->assertEquals('http://foo.com/', (string) $request->getUri());
+        $this->assertEquals('', $request->getHeaderLine('Content-Type'));
+    }
+
+    public function testPreparesRequestsWithJsonValueTraitArray()
+    {
+        $jsonValueArgs = [
+            "a" => "b"
+        ];
+        $request = $this->getRequest('foobar', ['baz' => $jsonValueArgs]);
+        $this->assertEquals('eyJhIjoiYiJ9', $request->getHeaderLine('baz'));
+        $this->assertEquals('POST', $request->getMethod());
+        $this->assertEquals('http://foo.com/', (string) $request->getUri());
+        $this->assertEquals('', $request->getHeaderLine('Content-Type'));
+    }
+
+    public function testPreparesRequestsWithJsonValueTraitEmptyString()
+    {
+        $jsonValueArgs = '';
+        $request = $this->getRequest('foobar', ['baz' => $jsonValueArgs]);
+        $this->assertEquals('IiI=', $request->getHeaderLine('baz'));
+        $this->assertEquals('POST', $request->getMethod());
+        $this->assertEquals('http://foo.com/', (string) $request->getUri());
+        $this->assertEquals('', $request->getHeaderLine('Content-Type'));
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
```

Comment:
Maybe this should be explicitly `\InvalidArgumentException`? I believe PHPUnit evaluates expected exceptions in the global namespace, but per the imports  this is actually expecting `\Doctrine\Instantiator\Exception\InvalidArgumentException`, which I assume your IDE imported automatically when searching for `InvalidArgumentException`;